Replicator/ConnectedClient/ConnectedClient.cc

```diff
@@ -141,7 +153,8 @@ namespace litecore::client {
                 }
                 gotError(C4Error::make(domain, code, status.message));
             }
-
+            
+            LOCK(_mutex);
```

Comment:
The scope of the lock shouldn't extend past the delegate call; wrap curly braces around these 3 lines.